### ISSUE TITLE
Query was already executed: modelhere.find({})

### ISSUE THREAD

**Issue by rosman83** [jeka-kiselyov/fastify-mongoose-api#9]
I am running into an issue generating CRUD models for a very basic model. All routes work as normal except using the GET method at the root of the model - which is intended to list all collection objects matching that model type.
```
    await server.register(fastifyMongooseAPI, {
        models: mongoose.models,
        prefix: '/api/',
        setDefaults: true,
        methods: ['list', 'get', 'post', 'patch', 'put', 'delete', 'options']
    });
```

```[1645364211632] INFO (13016 on DESKTOP-H0SUPE1): incoming request
    req: {
      "method": "GET",
      "url": "/api/servers",
      "hostname": "localhost:1600",
      "remoteAddress": "127.0.0.1",
      "remotePort": 52945
    }
    reqId: "10aa48ee-6f6c-44d0-XXX"
[1645364211639] ERROR (13016 on DESKTOP-H0SUPE1): Query was already executed: server.find({})
    MongooseError: Query was already executed: server.find({})
        at model.Query._wrappedThunk [as _find] (C:\XXX\node_modules\.pnpm\mongoose@6.2.2\node_modules\mongoose\lib\helpers\query\wrapThunk.js:21:19)
        at C:\xxx\node_modules\.pnpm\kareem@2.3.4\node_modules\kareem\index.js:372:33
        at processTicksAndRejections (node:internal/process/task_queues:78:11)
[1645364211641] INFO (13016 on DESKTOP-H0SUPE1): request completed
    res: {
      "statusCode": 500
    }
    responseTime: 7.673600003123283
    reqId: "XXX-8cd2b731b4b6"```

**Comment by senatet** [comment]
I ran into this issue too. I believe it is present in mongoose >= 6.0.0 (https://github.com/Automattic/mongoose/blob/master/CHANGELOG.md#600-rc0--2021-08-03). One of the breaking changes seems to be that queries cannot be re-executed. A simple fix in the code is to add clone() call to the APIRouter:getListResponse(query, request)  (https://github.com/jeka-kiselyov/fastify-mongoose-api/blob/master/src/APIRouter.js) `query.countDocuments()` call (i.e.: Change it to: `query.clone().countDocuments()`), and maybe also just generate a new query at the top of that function with:
`query = query.clone()` to ensure we are working with a freshly created query.

Either that, or maybe downgrade to mongoose < 6.0.0.

**Comment by jeka-kiselyov** [comment]
Hi @rosman83 @senatet 

Thank you for feedback and letting me know.

Issues fixed for mongoose > 6.00 in v 1.2.1: https://www.npmjs.com/package/fastify-mongoose-api 

Covered by tests, should work now
